securedrop_client/gui/widgets.py

```diff
@@ -19,181 +19,553 @@
+        self.login_button.setup(window)
+
+    def set_username(self, username):
+        self.user_icon.setText(_('jo'))
+        self.user_button.set_username(username)
+
+    def show(self):
+        self.login_button.hide()
+        self.user_icon.show()
+        self.user_button.show()
+
+    def hide(self):
+        self.user_icon.hide()
+        self.user_button.hide()
+        self.login_button.show()
+
+
+class UserButton(SvgPushButton):
+    """An menu button for the journalist menu
+
+    This button is responsible for launching the journalist menu on click.
+    """
+
+    css = '''
+    SvgPushButton#user_button {
+        border: none;
+        padding-left: 6px;
+        background-color: #0093da;
+        font-family: Open Sans;
+        font-size: 14px;
+        font-weight: bold;
+        color: #b4fffa;
+        align: left;
+        text-align: left;
+    }
+    SvgPushButton::menu-indicator {
+        image: none;
+    }
+    '''
+
+    def __init__(self):
+        super().__init__('dropdown_arrow.svg', svg_size=QSize())
+
+        self.setStyleSheet(self.css)
+        self.setFixedHeight(40)
+
+        self.setObjectName('user_button')
+
+        self.setLayoutDirection(Qt.RightToLeft)
 
-        Assign a controller object (containing the application logic) to this
-        instance of the toolbar.
+        self.menu = UserMenu()
+        self.setMenu(self.menu)
+
+    def setup(self, controller):
+        self.menu.setup(controller)
+
+    def set_username(self, username):
+        self.setText(_('{}').format(html.escape(username)))
+
+
+class UserMenu(QMenu):
+    """A menu next to the journalist username.
+
+    A menu that provides login options.
+    """
+
+    def __init__(self):
+        super().__init__()
+        self.logout = QAction(_('SIGN OUT'))
+        self.logout.setFont(QFont("OpenSans", 10))
+        self.addAction(self.logout)
+        self.logout.triggered.connect(self._on_logout_triggered)
+
+    def setup(self, controller):
+        """
+        Store a reference to the controller (containing the application logic).
         """
-        self.window = window
         self.controller = controller
 
-    def set_logged_in_as(self, username):
+    def _on_logout_triggered(self):
         """
-        Update the UI to reflect that the user is logged in as "username".
+        Called when the logout button is selected from the menu.
         """
-        self.login.hide()
+        self.controller.logout()
 
-        self.user_icon.setText(_('jo'))
-        self.user_icon.setStyleSheet('''
-            QLabel { background-color: #045fb4; color: cyan; padding: 10; border: 1px solid gray; }
-        ''')
-        self.user_icon.show()
 
-        self.user_state.setText(_('{}').format(html.escape(username)))
-        self.user_state.show()
+class LoginButton(QPushButton):
+    """
+    A button that opens a login dialog when clicked.
+    """
 
-        self.user_menu.show()
+    css = '''
```

Comment:
Nit: class constant should be `SCREAMING_SNAKE_CASE`

Reply:
lol good catch